stdlib/json.gr

```diff
@@ -0,0 +1,2155 @@
+      printNull(buffer)
+      return None
+    },
+    JsonBoolean(b) => {
+      printBool(b, buffer)
+      return None
+    },
+    JsonNumber(n) => return printNumber(n, buffer),
+    JsonString(s) => {
+      implHelper.emitEscapedQuotedStringPretty(s)
+      return None
+    },
+    JsonArray(elems) => {
+      match (elems) {
+        [] => {
+          Buffer.addChar('[', buffer)
+          Buffer.addChar(']', buffer)
+          return None
+        },
+        [e] => {
+          let format = implHelper.format
+
+          Buffer.addChar('[', buffer)
+
+          if (format.arrayFormat == OneArrayEntryPerLine) {
+            implHelper.printNewLine()
+          }
+
+          let elemLevel = indentationLevel + 1
+
+          if (format.arrayFormat == OneArrayEntryPerLine) {
+            implHelper.printIndentation(elemLevel)
+          }
+
+          let err = printElementPretty(e, implHelper, elemLevel)
+          if (Option.isSome(err)) return err
+
+          if (format.arrayFormat == OneArrayEntryPerLine) {
+            implHelper.printNewLine()
+            implHelper.printIndentation(indentationLevel)
+          }
+
+          Buffer.addChar(']', buffer)
+
+          return None
+        },
+        [initialHead, ...initialRest] => {
+          let format = implHelper.format
+
+          Buffer.addChar('[', buffer)
+
+          if (format.arrayFormat == OneArrayEntryPerLine) {
+            implHelper.printNewLine()
+          }
+
+          let mut currentHead = initialHead
+          let mut currentRest = initialRest
+
+          let elemLevel = indentationLevel + 1
+
+          for (let mut index = 0; ; index += 1) {
+            if (index > 0) {
+              Buffer.addChar(',', buffer)
+
+              if (format.arrayFormat == OneArrayEntryPerLine) {
+                implHelper.printNewLine()
+              }
+            }
+
+            if (format.arrayFormat == OneArrayEntryPerLine) {
+              implHelper.printIndentation(elemLevel)
+            }
+
+            let err = printElementPretty(currentHead, implHelper, elemLevel)
+            if (Option.isSome(err)) return err
+
+            match (currentRest) {
+              [] => break,
+              [newHead, ...newRest] => {
+                currentHead = newHead
+                currentRest = newRest
+              },
+            }
+          }
+
+          if (format.arrayFormat == OneArrayEntryPerLine) {
+            implHelper.printNewLine()
+            implHelper.printIndentation(indentationLevel)
+          }
+
+          Buffer.addChar(']', buffer)
+
+          return None
+        },
+      }
+    },
+    JsonObject(entries) => {
+      match (entries) {
+        [] => {
+          Buffer.addChar('{', buffer)
+          Buffer.addChar('}', buffer)
+          return None
+        },
+        [(key, value)] => {
+          let format = implHelper.format
+
+          Buffer.addChar('{', buffer)
+
+          let elemLevel = indentationLevel + 1
```

Comment:
I believe that if we make the JsonWriter work for compact and pretty, that we'll need a value indicating how much to increase the indentation level.

Reply:
Looking into this a bit more, no we do not need a value indicating how much as `printIndentation` handles the actual space generation. `indentationlevel` is calculating the number of indents to perform. which gets multiplied by the indentation in `printIndentation`.